scipy/linalg/_decomp_cossin.py

```diff
@@ -0,0 +1,186 @@
+from collections import Iterable
+
+import numpy as np
+from numpy.linalg import LinAlgError
+
+from . import block_diag
+from .lapack import _compute_lwork, get_lapack_funcs
+
+__all__ = ['cossin']
+
+from .._lib._util import _asarray_validated
+
+
+def cossin(X, partitioning=None):
+    """
+    Compute the cosine-sine (CS) decomposition of an M-by-M matrix.
+
+    The matrix X is a unitary or an orthogonal matrix, partitioned to X11, X12,
+    X21, X22 subblocks and the decomposition satisfies the following equality:
+
+    .. math::
```

Comment:
I would have appreciated the ascii version as they are often printed in ipython consoles rather than the actual docs. I think ascii was more readable anyways.

Reply:
yeah, I was just experimenting to see if this passes and then I left the ``` in 🤦‍♂. (I am working in little 30 minute slices these last couple of days). Also I wasn't sure what's the guidance regarding big formulas - I am glad to see and I agree too that the ascii version is more readable.